contrib/macdeploy/macdeployqtplus

```diff
@@ -543,6 +546,9 @@ ds.close()
 
 # ------------------------------------------------
 
+if need_to_sign_target and platform.system() == "Darwin" and platform.machine() == "arm64":
```

Comment:
Is there a problem with just always signing in the `need_to_sign_target` case,  i.e non-depends builds, rather than making this specific to arm64?

Reply:
Not a problem at all, just not required.

Reply:
Probably easier to just always sign then. Otherwise this should have an explainer.

Reply:
Is the following comment good as an explanation (compiled from Apple [docs](https://developer.apple.com/documentation/macos-release-notes/macos-big-sur-11_0_1-universal-apps-release-notes)):

_"In macOS 11 on Macs with Apple silicon the operating system enforces that any executable must be signed before it’s allowed to run. After using tools such as strip or install_name_tool, to call codesign is required to properly ad-hoc sign binaries."_

?

Reply:
I agree that adding a comment here makes sense. It's not obvious why this would be ARM-only.

Reply:
I am of the opinion that it is just easier to always sign. Right now the requirement is only for apple silicon, but I have not seen anything to say that this wouldn't expand to remaining intel Macs in future OS versions.

If we stick with only signing for arm, then the following is a suitable comment:

> "Starting with macOS 11 on Apple silicon, executables must be signed before they are allowed to run."

Reply:
> Probably easier to just always sign then. Otherwise this should have an explainer.

[Updated](https://github.com/bitcoin/bitcoin/pull/22546#issuecomment-919073042).